src/Microsoft.Azure.ServiceBus/Primitives/ServiceBusConnection.cs

```diff
@@ -51,7 +51,7 @@ public Task CloseAsync()
 
         protected void InitializeConnection(ServiceBusConnectionStringBuilder builder)
         {
-            this.Endpoint = builder.Endpoint;
+            this.Endpoint = new Uri(builder.Endpoint);
```

Comment:
Sure we could have a custom type for lot of things, but IMHO, that separate type won't have much more functionality than what Uri class provides.